src/lib/autoDetect.ts

```diff
@@ -0,0 +1,73 @@
+import { ExtensionContext, workspace, window, languages } from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  languages.setLanguageConfiguration('hubl', {
+    indentationRules: {
+      decreaseIndentPattern: /.*{%(.*?end).*%}*./,
+      increaseIndentPattern: /.*{%(?!.*end).*%}.*/,
+    },
+  });
```

Comment:
We have HubL language configured in `./langconfig/hubl-language-configuration.json` -- I don't believe this is needed now that we have those configuration files (it looks like Anthony's PR is from a while ago and was pre-lang configs being fleshed out so he was putting some configuration rules in his code directly)

```suggestion
```